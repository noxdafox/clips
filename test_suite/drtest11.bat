(clear)                   ; Optimized Rete Evaluator Issue

(defrule rule-1
   (factoid ?x ?y&:(and ?x ?y)) ; FactPNGetVar3
   =>)

(defrule rule-2
   (factoid ?x ?y)
   (test (and ?x ?y)) ; FactJNGetVar3
   =>)

(defrule rule-3
   (factoid $? ?x ?y&:(and ?x ?y) $?) ; FactPNGetVar1
   =>)

(defrule rule-4
   (factoid $? ?x ?y $?)
   (test (and ?x ?y)) ; FactJNGetVar1
   =>)
(defglobal ?*z* = FALSE)

(defrule rule-5
   (factoid ? ?)
   (test (and ?*z* ?*z*))
   =>)
(assert (factoid FALSE FALSE))
(agenda)
(clear)

(deftemplate factoid
   (slot s1)
   (slot s2))

(defrule rule-1
   (factoid (s1 ?x) (s2 ?y&:(and ?x ?y))) ; FactPNGetVar2
   =>)

(defrule rule-2
   (factoid (s1 ?x) (s2 ?y))
   (test (and ?x ?y)) ; FactJNGetVar2
   =>)
(assert (factoid (s1 FALSE) (s2 FALSE)))
(agenda)
(clear)

(defclass OBJOID1
   (is-a USER)
   (slot s1)
   (slot s2))

(defclass OBJOID2
   (is-a USER)
   (multislot ms1))

(defrule rule-1
   (object 
      (is-a OBJOID1)
      (s1 ?x)
      (s2 ?y&:(and ?x ?y))) ; ObjectGetVarPNFunction1
   =>)

(defrule rule-2
   (object 
      (is-a OBJOID1)
      (s1 ?x)
      (s2 ?y))
   (test (and ?x ?y)) ; ObjectGetVarJNFunction1
   =>)

(defrule rule-3
   (object 
      (is-a OBJOID2)
      (ms1 $? ?x ?y&:(and ?x ?y))) ; ObjectGetVarPNFunction2
   =>)
   
(defrule rule-4
   (object 
      (is-a OBJOID2)
      (ms1 $? ?x ?y))
   (test (and ?x ?y)) ; ObjectGetVarJNFunction2
   =>)
(make-instance o1 of OBJOID1 (s1 FALSE) (s2 FALSE))
(make-instance o2 of OBJOID2 (ms1 FALSE FALSE))
(agenda)
(clear)
