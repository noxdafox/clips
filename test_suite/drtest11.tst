(unwatch all)
(clear)
(set-strategy depth)
(open "Results//drtest11.rsl" drtest11 "w")
(dribble-on "Actual//drtest11.out")
(batch "drtest11.bat")
(dribble-off)
(load "compline.clp")
(printout drtest11 "drtest11.bat differences are as follows:" crlf)
(compare-files "Expected//drtest11.out" "Actual//drtest11.out" drtest11)
; close result file
(close drtest11)
