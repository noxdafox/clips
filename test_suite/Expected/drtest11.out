TRUE
CLIPS> (batch "drtest11.bat")
TRUE
CLIPS> (clear)                   ; Optimized Rete Evaluator Issue
CLIPS> 
(defrule rule-1
   (factoid ?x ?y&:(and ?x ?y)) ; FactPNGetVar3
   =>)
CLIPS> 
(defrule rule-2
   (factoid ?x ?y)
   (test (and ?x ?y)) ; FactJNGetVar3
   =>)
CLIPS> 
(defrule rule-3
   (factoid $? ?x ?y&:(and ?x ?y) $?) ; FactPNGetVar1
   =>)
CLIPS> 
(defrule rule-4
   (factoid $? ?x ?y $?)
   (test (and ?x ?y)) ; FactJNGetVar1
   =>)
CLIPS> (defglobal ?*z* = FALSE)
CLIPS> 
(defrule rule-5
   (factoid ? ?)
   (test (and ?*z* ?*z*))
   =>)
CLIPS> (assert (factoid FALSE FALSE))
<Fact-1>
CLIPS> (agenda)
CLIPS> (clear)
CLIPS> 
(deftemplate factoid
   (slot s1)
   (slot s2))
CLIPS> 
(defrule rule-1
   (factoid (s1 ?x) (s2 ?y&:(and ?x ?y))) ; FactPNGetVar2
   =>)
CLIPS> 
(defrule rule-2
   (factoid (s1 ?x) (s2 ?y))
   (test (and ?x ?y)) ; FactJNGetVar2
   =>)
CLIPS> (assert (factoid (s1 FALSE) (s2 FALSE)))
<Fact-1>
CLIPS> (agenda)
CLIPS> (clear)
CLIPS> 
(defclass OBJOID1
   (is-a USER)
   (slot s1)
   (slot s2))
CLIPS> 
(defclass OBJOID2
   (is-a USER)
   (multislot ms1))
CLIPS> 
(defrule rule-1
   (object 
      (is-a OBJOID1)
      (s1 ?x)
      (s2 ?y&:(and ?x ?y))) ; ObjectGetVarPNFunction1
   =>)
CLIPS> 
(defrule rule-2
   (object 
      (is-a OBJOID1)
      (s1 ?x)
      (s2 ?y))
   (test (and ?x ?y)) ; ObjectGetVarJNFunction1
   =>)
CLIPS> 
(defrule rule-3
   (object 
      (is-a OBJOID2)
      (ms1 $? ?x ?y&:(and ?x ?y))) ; ObjectGetVarPNFunction2
   =>)
CLIPS>    
(defrule rule-4
   (object 
      (is-a OBJOID2)
      (ms1 $? ?x ?y))
   (test (and ?x ?y)) ; ObjectGetVarJNFunction2
   =>)
CLIPS> (make-instance o1 of OBJOID1 (s1 FALSE) (s2 FALSE))
[o1]
CLIPS> (make-instance o2 of OBJOID2 (ms1 FALSE FALSE))
[o2]
CLIPS> (agenda)
CLIPS> (clear)
CLIPS> (dribble-off)
